### PR TITLE
finish program choices, untested

### PR DESCRIPTION
**Pull request by lucy-fan**
please look at **choices.service.ts**, rest is misc code from previous weeks that need to be fixed, but not now :)

### CODE REVIEW

File: application-portal/src/app/models/choices/choices.ts [hunk 1]
```diff
@@ -1,7 +1,10 @@
 export class Choices {
```
Comment (khannotations) [inline application-portal/src/app/models/choices/choices.ts]: Model names should be singular, i.e. this should be `Choice` or better `ProgramChoice`

File: application-portal/src/app/services/choices/choices.service.ts [hunk 2]
```diff
@@ -1,9 +1,284 @@
...
-  constructor() { }
-}
+  // arrays that hold info from backend
+  programs: Array<Program>;
+  locations: Array<Location>;
+  periods: Array<Period>;
+  sessions: Array<Session>;
+
+  // fields arrays
+  tech: Array<string> = ["Data Science Course", "Full Stack Coding Course", "Digital Design Course"];
```
Comment (khannotations) [inline application-portal/src/app/services/choices/choices.service.ts]: Nice

File: application-portal/src/app/services/choices/choices.service.ts [hunk 3]
```diff
@@ -1,9 +1,284 @@
...
+  program: Program;
+  location: Location;
+  period: Period;
+  session: Session;
+  choices: Choices;
 
-  constructor() { }
-}
+  // arrays that hold info from backend
+  programs: Array<Program>;
```
Comment (khannotations) [inline application-portal/src/app/services/choices/choices.service.ts]: you might consider naming these a bit more explicitly so you don't get confused, i.e. `allPrograms`, `allLocations`, etc.

File: application-portal/src/app/services/choices/choices.service.ts [hunk 4]
```diff
@@ -1,9 +1,284 @@
...
+  */
+
+  /* Basic Get/Set for all of the fields in Choices, UNTESTED */
+
+  getProgram() {
+    return this.program;
+  }
+
+  setProgram(program_id: number) {
+    for (var i = 0; i < this.programs.length; i++) {
```
Comment (khannotations) [inline application-portal/src/app/services/choices/choices.service.ts]: Why do you need to loop through the programs if you're given the program id already?

File: application-portal/src/app/services/choices/choices.service.ts [hunk 5]
```diff
@@ -1,9 +1,284 @@
...
+      if (this.programs[i].id == program_id) {
+        // set choices' program id
+        this.choices.program_id = this.programs[i].id;
+        // set program variable 
+        this.program = this.programs[i];
```
Comment (khannotations) [inline application-portal/src/app/services/choices/choices.service.ts]: `Array.find` is a cleaner way to search an array for something (without having to loop through it yourself)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

**Comment by lucy-fan** [comment]
changes have been added! please take a look when you have time :)

File: application-portal/src/app/models/choices/choices.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-export class Choices {
+export class ProgramChoice {
```
Comment (khannotations) [inline application-portal/src/app/models/choices/choices.ts]: @lucy-fan You also need to rename this file.

**Pull request merged**